pkg/k8s/service_cache.go

```diff
@@ -420,6 +426,54 @@ func (s *ServiceCache) UniqueServiceFrontends() FrontendList {
 	return uniqueFrontends
 }
 
+// filterEndpoints filters local endpoints by using k8s service heuristics.
```

Comment:
Add more notes here about the k8s service heuristics. 

>Sometimes the controller allocates endpoints from a different zone to ensure more even distribution of endpoints between zones. This would result in some traffic being routed to other zones.

Specifically, if there are no topologically-close zones, then traffic would be routed to endpoints from other zones.

Reply:
Added the comment to the return which returns the whole set of endpoints if the filtered endpoints list == 0.